core/common/src/main/java/alluxio/underfs/UnderFileSystemFactoryRegistry.java

```diff
@@ -154,17 +154,26 @@ public static UnderFileSystemFactory find(
    *
    * @param path path
    * @param ufsConf configuration of the UFS
+   * @param useCache if true, will cache and reuse factory instances
    * @return list of factories that support the given path which may be an empty list
    */
   public static List<UnderFileSystemFactory> findAll(
-      String path, UnderFileSystemConfiguration ufsConf) {
+      String path, UnderFileSystemConfiguration ufsConf, boolean useCache) {
```

Comment:
will `useCache` ever be true after your change? if not, can we remove the static variables?

Reply:
in fact, if `LOADED_LIB_JARS` and other global state are never used, then I don't see why we still keep `UnderFileSystemFactoryRegistry`

Reply:
Yeah, I removed that option. It will never be cached now.

I think the class is useful for tests. The only factory that is "built-in" is the local UFS one, and the new ones that test cases register.

Reply:
@gpang can you clarify this in the javadoc of `FACTORIES`?

Reply:
I updated the javadoc.